src/inference/inference.ts

```diff
@@ -140,42 +140,50 @@ export class Inference {
           ' documents. See https://docs.pinecone.io/models for supported models.'
       );
     }
-    // Destructure `options` with defaults
-    // Note: If the user passes in key:value pairs in `options` that are not the following, they are ignored
+
     const {
       topN = documents.length,
       returnDocuments = true,
-      rankFields = ['text'],
       parameters = {},
+      rankFields = undefined,
     } = options;
 
-    // Allow documents to be passed a list of strings, or a list of objs w/at least a `text` key:
-    let newDocuments: Array<{ [key: string]: string }> = [];
-    if (typeof documents[0] === 'object' && !('text' in documents[0])) {
-      throw new PineconeArgumentError(
-        '`documents` can only be a list of strings or a list of objects with at least a `text` key, followed by a' +
-          ' string value'
-      );
-    } else if (typeof documents[0] === 'string') {
-      newDocuments = documents.map((doc) => {
-        return { text: doc as string };
-      });
+    let computedRankFields: string[];
+
+    // If user does not pass in rankFields:
+    if (!rankFields) {
+      // If `documents` is an array of strings:
+      if (documents.every((doc) => typeof doc === 'string')) {
+        // Set rankFields to default value of 'text'
+        computedRankFields = ['text'];
+      } else if (
+        // If`documents` is an array of objects:
+        documents.every((doc) => typeof doc === 'object' && doc !== null)
+      ) {
+        // If there is a `text` key (other keys can be present too):
+        if (
+          documents.every((doc) => (doc as { [key: string]: string })['text'])
+        ) {
+          computedRankFields = ['text'];
+        } else {
+          // If there is no `text` key:
+          throw new PineconeArgumentError(
+            'One or more documents are missing the specified rank field: `text`'
+          );
+        }
+      } else {
+        throw new PineconeArgumentError(
+          'Documents must be a list of strings or a list of objects'
+        );
+      }
     } else {
-      newDocuments = documents as Array<{ [key: string]: string }>;
+      computedRankFields = rankFields;
     }
 
-    // Ensure all rankFields, if passed, are present in each document
-    if (rankFields.length > 0) {
-      newDocuments.forEach((doc, index) => {
-        rankFields.forEach((field) => {
-          if (!(field in doc)) {
-            throw new PineconeArgumentError(
-              `The \`rankField\` value you passed ("${field}") is missing in the document at index ${index}`
-            );
-          }
-        });
-      });
-    }
+    // Standardize documents to ensure they are in object format
+    const newDocuments = documents.map((doc) =>
+      typeof doc === 'string' ? { text: doc } : doc
```

Comment:
Will the `text` key here be different if they passed a custom `rankFields`?

Reply:
I don't think that's a valid scenario, here's my thinking: 
- If someone passes in an array of strings, there are no keys in their documents
- If there are no keys in their documents, on the backend we artificially transform their documents into an array of key:value pairs, with a `text` key, simply b/c that's what the model needs for reranking
- If someone passes in an array of strings and then also passes in a custom field to rerank on... their docs would inherently be missing that custom field (by way of not being key:value pairs), so it would error out

Thoughts @ssmith-pc ?

Reply:
I'm not fully following the original question since the impl has changed, but:

> * If there are no keys in their documents, on the backend we artificially transform their documents into an array of key:value pairs, with a text key, simply b/c

By "on the backend we transform" I think you mean, it's transformed in the client right? This is the right behavior

> * If someone passes in an array of strings and then also passes in a custom field to rerank on... their docs would inherently be missing that custom field (by way of not being key:value pairs), so it would error out

Correct, but it's fine if that error is detected in the server and sent back to the client. The client doesn't have to try and detect that.